src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -46,100 +65,159 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 		});
 	}
 
-	private async loadBtcTransactionsData({ btcAddress }: { btcAddress: BtcAddress }): Promise<{
-		newTransactions: BitcoinTransaction[];
-		latestBitcoinBlockHeight: number | undefined;
-	}> {
+	private async loadBtcTransactionsData({
+		btcAddress
+	}: {
+		btcAddress: BtcAddress;
+	}): Promise<CertifiedData<BtcTransactionUi>[]> {
 		try {
 			const { txs: fetchedTransactions } = await btcAddressData({ btcAddress });
 
 			const newTransactions = fetchedTransactions.filter(({ hash }) =>
 				isNullish(this.store.transactions[`${hash}`])
 			);
 
-			this.store = {
-				...this.store,
-				transactions: {
-					...this.store.transactions,
-					...newTransactions.reduce(
-						(acc, transaction) => ({
-							...acc,
-							[transaction.hash]: {
-								certified: false,
-								data: transaction
-							}
-						}),
-						{}
-					)
-				}
-			};
-
 			const latestBitcoinBlockHeight = await btcLatestBlockHeight();
 
-			return { newTransactions, latestBitcoinBlockHeight };
+			return nonNullish(latestBitcoinBlockHeight)
+				? newTransactions.map((transaction) => ({
+						data: mapBtcTransaction({ transaction, btcAddress, latestBitcoinBlockHeight }),
+						certified: false
+					}))
+				: [];
 		} catch (error) {
 			// We don't want to disrupt the user experience if we can't fetch the transactions or latest block height.
 			console.error('Error fetching BTC transactions data:', error);
-			// TODO: Return an error instead of an object with empty array.
-			return { newTransactions: [], latestBitcoinBlockHeight: undefined };
+			// TODO: Return an error instead of an empty array.
+			return [];
 		}
 	}
 
-	private async loadBtcBalance({
+	private loadBtcBalance = async ({
 		identity,
-		bitcoinNetwork
-	}: {
-		identity: OptionIdentity;
-		bitcoinNetwork: BitcoinNetwork;
-	}): Promise<CertifiedData<bigint>> {
-		const balance = await getBtcBalance({
-			identity,
-			network: bitcoinNetwork,
-			minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
-		});
-		const certifiedBalance = {
-			data: balance,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		certified = true
+	}: Omit<LoadBtcWalletParams, 'shouldFetchTransactions'>): Promise<
+		CertifiedData<bigint> | undefined
```

Comment:
Is it correct to return `undefined`? In what case is the balance undefined? And shouldn't the return be `{balance: undefined, certified: false}` or is there a case where the balance is undefined and certified? And should we use `null` to express the fact that it's a choice?

Reply:
Since `isNullish(this.store.balance)` is now gone, balance can be nullish only in a very specific case - when minter and/or BTC canister ids are not available. I left a comment in the code with a hint on this corner case. Also, updated the value to `null`.